packages/fxa-settings/src/pages/ResetPasswordRedesign/CompleteResetPasswordOtp/container.tsx

```diff
@@ -0,0 +1,207 @@
+import { getLocalizedErrorMessage } from '../../../lib/auth-errors/auth-errors';
+import { useState } from 'react';
+
+// This component is used for both /complete_reset_password and /account_recovery_reset_password routes
+// for easier maintenance
+
+const CompleteResetPasswordOtpContainer = ({
+  integration,
+}: {
+  integration: Integration;
+} & RouteComponentProps) => {
+  const keyStretchExperiment = useValidatedQueryParams(KeyStretchExperiment);
+
+  const account = useAccount();
+  const config = useConfig();
+  const ftlMsgResolver = useFtlMsgResolver();
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const searchParams = location.search;
+
+  if (!location.state) {
+    navigate(`/reset_password${searchParams}`);
+  }
+
+  const {
+    code,
+    email,
+    token,
+    accountResetToken,
+    emailToHashWith,
+    kB,
+    recoveryKeyId,
+  } = location.state as CompleteResetPasswordLocationState;
+
+  const hasConfirmedRecoveryKey = !!(
+    accountResetToken &&
+    email &&
+    kB &&
+    recoveryKeyId
+  );
+
+  const isResetWithoutRecoveryKey = !!(code && token);
+
+  // TODO in FXA-7890 handle case where hasAccountRecoveryKey could not be determined
+  // and show banner to use account recovery key
+  // add metrics to track if users ever click on the link
+
+  const handleNavigationWithRecoveryKey = () => {
+    navigate(`/reset_password_with_recovery_key_verified${location.search}`, {
+      replace: true,
+    });
+  };
+
+  const handleNavigationWithoutRecoveryKey = () => {
+    navigate(`/reset_password_verified${searchParams}`, {
+      replace: true,
+    });
+  };
+
+  const resetPasswordWithRecoveryKey = async (
+    accountResetToken: string,
+    emailToUse: string,
+    kB: string,
+    newPassword: string,
+    recoveryKeyId: string
+  ) => {
+    const options = {
+      accountResetToken,
+      emailToHashWith: emailToUse,
+      password: newPassword,
+      recoveryKeyId,
+      kB,
+    };
+
+    // TODO move out of account model?
+    // Reset with recovery key directly uses authClient instead of GQL
+    const accountResetData = await account.resetPasswordWithRecoveryKey(
+      options
+    );
+
+    return accountResetData;
+  };
+
+  const resetPasswordWithoutRecoveryKey = async (
+    code: string,
+    emailToUse: string,
+    newPassword: string,
+    token: string
+  ) => {
+    // TODO move out of account model?
+    const accountResetData: AccountResetData =
+      await account.completeResetPassword(
+        keyStretchExperiment.queryParamModel.isV2(config),
+        token,
+        code,
+        emailToUse,
+        newPassword,
+        accountResetToken
+      );
+    return accountResetData;
+  };
+
+  const notifyBrowserOfSignin = async (accountResetData: AccountResetData) => {
+    if (
+      isSyncDesktopV3Integration(integration) ||
+      isSyncOAuthIntegration(integration)
```

Comment:
Instead of `isSyncDesktopV3Integration(integration) || isSyncOAuthIntegration(integration)`, we should use `integration.isSync()`. The OAuth integration is not always sync.